kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ApplyMojo.java

```diff
@@ -184,58 +176,15 @@ public void executeInternal() throws MojoExecutionException {
                 clusterKind = "OpenShift";
             }
             KubernetesResourceUtil.validateKubernetesMasterUrl(masterUrl);
-            log.info("Using %s at %s in namespace %s with manifest %s ", clusterKind, masterUrl, clusterAccess.getNamespace(), manifest);
-
-            applyService.setAllowCreate(createNewResources);
-            applyService.setServicesOnlyMode(servicesOnly);
-            applyService.setIgnoreServiceMode(ignoreServices);
-            applyService.setLogJsonDir(jsonLogDir);
-            applyService.setBasedir(getRootProjectFolder());
-            applyService.setIgnoreRunningOAuthClients(ignoreRunningOAuthClients);
-            applyService.setProcessTemplatesLocally(processTemplatesLocally);
-            applyService.setDeletePodsOnReplicationControllerUpdate(deletePodsOnReplicationControllerUpdate);
-            applyService.setRollingUpgrade(rollingUpgrades);
-            applyService.setRollingUpgradePreserveScale(isRollingUpgradePreserveScale());
-            applyService.setRecreateMode(recreate);
-
-            boolean openShift = OpenshiftHelper.isOpenShift(kubernetes);
-            if (openShift) {
-                log.info("[[B]]OpenShift[[B]] platform detected");
-            } else {
-                disableOpenShiftFeatures(applyService);
-            }
-
             Set<HasMetadata> entities = KubernetesResourceUtil.loadResources(manifest);
+            log.info("Using %s at %s in namespace %s with manifest %s ", clusterKind, masterUrl, clusterAccess.getNamespace(), manifest);
 
-            String namespace = clusterAccess.getNamespace();
-            boolean namespaceEntityExist = false;
-
-            for (HasMetadata entity: entities) {
-                if (entity instanceof Namespace) {
-                    Namespace ns = (Namespace)entity;
-                    namespace = ns.getMetadata().getName();
-                    applyService.applyNamespace((ns));
-                    namespaceEntityExist = true;
-                    entities.remove(entity);
-                    break;
-                }
-                if (entity instanceof Project) {
-                    Project project = (Project)entity;
-                    namespace = project.getMetadata().getName();
-                    applyService.applyProject(project);
-                    namespaceEntityExist = true;
-                    entities.remove(entity);
-                    break;
-                }
-            }
-
-            if (!namespaceEntityExist) {
-                applyService.applyNamespace(namespace);
-            }
+            configureApplyService(kubernetes, entities);
 
-            applyService.setNamespace(namespace);
+            applyService.applyNamespaceOrProjectIfPresent(entities);
```

Comment:
Sorry, but I'm still unsure why this method is not part of `applyService#applyEntities` logic.
Why do we have to call it separately?
Is it something that applies specifically to Maven projects, or is this something we want for the Kit API?

Reply:
It was added here to ensure that any `Namespace`/`Project` is applied before any resource. It was there in the Mojo already; I just moved logic to a method. I think you're right. It should be called from `applyEntities()` method.

Reply:
Anyway, whatever it was supposed to do seems broken. I've created https://github.com/eclipse/jkube/issues/511 for `Namespace` fragment creation issues. We might need to modify `DefaultNamespaceEnricher` a bit.